azurelinuxagent/common/protocol/util.py

```diff
@@ -63,16 +64,14 @@ def get_protocol_util():
     return ProtocolUtil()
 
 
-class ProtocolUtil(object):
+class ProtocolUtil(SingletonPerThread):
     """
     ProtocolUtil handles initialization for protocol instance. 2 protocol types
     are invoked, wire protocol and metadata protocols.
     """
 
     def __init__(self):
-        self.lock = threading.Lock()
-        self.lock_wireserver_endpoint = threading.Lock()
-        self.protocol = None
+        self._protocol = None
```

Comment:
Made the protocol object private to discourage people from using the property directly and always go through get_protocol()